pkg/db/db_releases.go

```diff
@@ -534,15 +547,17 @@ func (h *DBHandler) insertReleaseHistoryRow(ctx context.Context, transaction *sq
 	return nil
 }
 
-func (h *DBHandler) DBMigrationUpdateReleasesTimestamp(ctx context.Context, transaction *sql.Tx, application string, releaseversion types.ReleaseNumbers, createAt time.Time) error {
-	span, ctx, onErr := tracing.StartSpanFromContext(ctx, "DBMigrationUpdateReleasesTimestamp")
-	defer span.Finish()
+func (h *DBHandler) DBMigrationUpdateReleasesTimestamp(ctx context.Context, transaction *sql.Tx, application string, releaseversion types.ReleaseNumbers, createAt time.Time) (err error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBMigrationUpdateReleasesTimestamp")
+	defer func() {
+		span.Finish(tracer.WithError(err))
+	}()
```

Comment:
please split this up into 3 functions, for example:
1: DBMigrationUpdateReleasesTimestamp
2: DBUpdateReleasesHistory
3: DBUpdateRelease

The outer function DBMigrationUpdateReleasesTimestamp should be quite simple, something like
err := DBUpdateReleasesHistory(...)
if (err...)
return DBUpdateRelease(...)